lib/web_ui/test/text/font_collection_test.dart

```diff
@@ -21,122 +21,192 @@ void main() {
+
+        expect(fontFamilyList.length, equals(2));
+        expect(fontFamilyList, contains('\'Ahem!!ahem\''));
+        expect(fontFamilyList, contains('Ahem!!ahem'));
       });
 
-      expect(fontFamilyList.length, equals(2));
-      expect(fontFamilyList, contains('\'/Ahem\''));
-      expect(fontFamilyList, contains('/Ahem'));
-    }, skip: (browserEngine == BrowserEngine.firefox));
+      test('Register Asset twice with comma', () async {
+        final String _testFontFamily = 'Ahem ,ahem';
+        final List<String> fontFamilyList = List<String>();
+
+        fontManager.registerAsset(
+            _testFontFamily, 'url($_testFontUrl)', const <String, String>{});
+        await fontManager.ensureFontsLoaded();
+        html.document.fonts
+            .forEach((html.FontFace f, html.FontFace f2, html.FontFaceSet s) {
+          fontFamilyList.add(f.family);
+        });
+
+        expect(fontFamilyList.length, equals(2));
+        expect(fontFamilyList, contains('\'Ahem ,ahem\''));
+        expect(fontFamilyList, contains('Ahem ,ahem'));
+      });
 
-    test('Register Asset twice with exclamation mark', () async {
-      final String _testFontFamily = 'Ahem!!ahem';
-      final List<String> fontFamilyList = List<String>();
+      test('Register Asset twice with a digit at the start of a token',
+          () async {
+        final String testFontFamily = 'Ahem 1998';
+        final List<String> fontFamilyList = List<String>();
+
+        fontManager.registerAsset(
+            testFontFamily, 'url($_testFontUrl)', const <String, String>{});
+        await fontManager.ensureFontsLoaded();
+        html.document.fonts
+            .forEach((html.FontFace f, html.FontFace f2, html.FontFaceSet s) {
+          fontFamilyList.add(f.family);
+        });
+
+        expect(fontFamilyList.length, equals(2));
+        expect(fontFamilyList, contains('Ahem 1998'));
+        expect(fontFamilyList, contains('\'Ahem 1998\''));
+      });
+    }, skip: (browserEngine == BrowserEngine.firefox));
 
-      fontManager.registerAsset(
-          _testFontFamily, 'url($_testFontUrl)', const <String, String>{});
-      await fontManager.ensureFontsLoaded();
-      html.document.fonts
-          .forEach((html.FontFace f, html.FontFace f2, html.FontFaceSet s) {
-        fontFamilyList.add(f.family);
+    group('special characters Firefox', () {
+      test('Register Asset with special character slash', () async {
+        final String _testFontFamily = '/Ahem';
+        final List<String> fontFamilyList = List<String>();
+
+        fontManager.registerAsset(
+            _testFontFamily, 'url($_testFontUrl)', const <String, String>{});
+        await fontManager.ensureFontsLoaded();
+        html.document.fonts
+            .forEach((html.FontFace f, html.FontFace f2, html.FontFaceSet s) {
+          fontFamilyList.add(f.family);
+        });
+
+        expect(fontFamilyList.length, equals(1));
+        expect(fontFamilyList.first, '\"/Ahem\"');
```

Comment:
Are the expectations the only difference between this test and the non-Firefox? If so, I think I'd prefer that we write one test and use different `expect`s, e.g.:

```dart
test('...', () async {
  ... shared test code ...
  if (browserEngine == firefox) {
    ... FF expectations ...
  } else {
    ... other expectations ...
  }
});
```

This way we won't forget to run tests in some browsers.

Reply:
Thanks, I merged the PR in the early morning. I'll address this in with another PR.

Reply:
sent this one out: https://github.com/flutter/engine/pull/14487